library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java

```diff
@@ -312,7 +315,7 @@ private void setupBitmapLayout() {
     float anchorX = parentLeft + (parentWidth * cuePosition);
     float anchorY = parentTop + (parentHeight * cueLine);
     int width = Math.round(parentWidth * cueSize);
-    int height = Math.round(width * ((float) cueBitmap.getHeight() / cueBitmap.getWidth()));
+    int height = Math.round(width * ((float) cueBitmap.getHeight() / cueBitmap.getWidth()) / (((float) parentWidth / parentHeight) / cueSar));
```

Comment:
I got a bit confused trying to work out exactly what SAR was :). In any case, looking at what this does, it feels like it would be simpler if Cue were to instead have the following field:
```
  /**
   * The bitmap height as a fraction of the of the viewport size, or -1 if the bitmap should be
   * displayed at its natural height given for its specified {@link #size}.
   */
  public final float bitmapHeight;
```
Then here you can do:
```
int height = cueBitmapHeight != -1 ? Math.round(parentWidth * cueBitmapHeight)
    : [what-was-here-before];
```